lib/src/platform_client.dart

```diff
@@ -837,6 +858,32 @@ class PlatformClient {
     await _httpPost('/api/smartapp/token', body);
   }
 
+  /// Requests Platform if there is an available Site Offer at the current position.
+  ///
+  /// Returns null if there is no Site Offer available or an `AccessOfferDetails` of the available offer if within
+  /// geofence.
+  Future<AccessOfferDetails?> inquireForGPSActivatedOffer(Position position) async {
+    _verifyIsLoggedIn();
+
+    if (null != _lastPositionUpdate && _lastPositionUpdate!.timeSinceInMs() < 1000) {
+      // Same throttling delay as `KurrentoRoom.updateLocation`.
+      return _lastAccessOfferUpdate;
+    }
+    _lastPositionUpdate = position;
+
+    String body = jsonEncode({
+      'userId': _userId,
+      'lt': position.latitude,
+      'lg': position.longitude,
+      'p': Platform.isAndroid ? 'ANDROID' : 'IOS',
+    });
+
+    Map<String, dynamic> gpsResponse = await _httpPost('/api/user/location', body);
```

Comment:
Can you please double check that `X-Device-Id` header is set? See the platform [API](https://github.com/aira/platform/blob/269f308197299fb6ce95d27ba396a54381bdf010/core/server/AiraPlatform/src/main/java/io/aira/rest/UserRestController.java#L113).

Reply:
yes it does! all the _http[post, get, delete] function do by default.